packages/playwright-core/src/client/elementHandle.ts

```diff
@@ -282,16 +282,16 @@ export async function convertInputFiles(files: string | FilePayload | string[] |
     if (!items.every(item => typeof item === 'string'))
       throw new Error('File paths cannot be mixed with buffers');
 
-    const [localPaths, localDirectory] = await resolvePathsAndDirectoryForInputFiles(items as string[]);
+    const [localPaths, localDirectory] = await resolvePathsAndDirectoryForInputFiles(items);
```

Comment:
This [is one of the highlights](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-5.html#inferred-type-predicates) in this TypeScript v5.5 release.